# This patch file was generated by NetBeans IDE
# Following Index: paths are relative to: /home/mkleint/src/winp/winp-1.13
# This patch can be applied using context Tools: Patch action on respective folder.
# It uses platform neutral UTF-8 encoding and \n newlines.
# Above lines and this line are ignored by the patching process.
Index: src/main/java/org/jvnet/winp/Native.java
--- src/main/java/org/jvnet/winp/Native.java Base (BASE)
+++ src/main/java/org/jvnet/winp/Native.java Locally Modified (Based On LOCAL)
@@ -45,6 +45,8 @@
     native static void noop();
 
     private static final Logger LOGGER = Logger.getLogger(Native.class.getName());
+    // system property holding the preferred folder for copying the dll file to.
+    private static final String DLL_TARGET = "winp.folder.preferred";
 
     static {
         load();
@@ -70,7 +72,8 @@
                     if(filePortion.startsWith("//"))
                         filePortion = filePortion.substring(2);
                     filePortion = URLDecoder.decode(filePortion);
-                    File jarFile = new File(filePortion);
+                    String preferred = System.getProperty(DLL_TARGET);
+                    File jarFile = new File(preferred != null ? preferred : filePortion);
                     File dllFile = new File(jarFile.getParentFile(),dllName+".dll");
                     if(!dllFile.exists() || jarFile.lastModified()>dllFile.lastModified()) {
                         // try to extract from within the jar
